### PR TITLE
docs: Add Keptn versioning and version compatibility document

### PR DESCRIPTION
**Pull request by thisthat**
Signed-off-by: Giovanni Liva <giovanni.liva@dynatrace.com>

### CODE REVIEW

**Review by agrimmer — APPROVED** [review]
lgtm

File: VERSIONING.MD [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Compatibility requirements
+
+This codebase marks all artifacts with a version of the format `MAJOR.MINOR.PATCH` or
+`MAJOR.MINOR.PATCH-next.ID`. For any artifact with a stable release, that is its version does not
+end in `-next.ID`, no backwards-incompatible changes will be made unless incrementing the `MINOR`
+version number. Patch releases with urgent cherry-picked bugfixes will be made by incrementing the
+`PATCH` version.
+
+A backwards-incompatible change affects the public API. The public API is any public reachable endpoint
+that is not marked as `internal`. We refer to a backward-incompatible API change, a change that could
```
Comment (mowies) [inline VERSIONING.MD:12]: Nitpick:
```suggestion
that is not marked as `internal`. We refer to a backwards-incompatible API change, a change that could
```

File: VERSIONING.MD [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+potentially cause failures in the applications that consume that API, e.g., removing an endpoint,
+changing the response code, requiring additional parameters, or changing the entity models.
+
+Such changes will be avoided - if they must be made, the affected endpoint will be marked as `deprecated`
+and supported for one further `MINOR` version. After that, the affected API may be made backward-incompatible.
```
Comment (mowies) [inline VERSIONING.MD:17]: Nitpick:
```suggestion
and supported for one further `MINOR` version. After that, the affected API may be made backwards-incompatible.
```

File: VERSIONING.MD [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Backwards incompatible changes to `internal` API are to be expected.
```
Comment (mowies) [inline VERSIONING.MD:19]: Nitpick:
```suggestion
Backwards-incompatible changes to `internal` API are to be expected.
```

**Review by mowies — APPROVED** [review]

**Review by mowies — APPROVED** [review]

File: VERSIONING.MD [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## Compatibility requirements
+
+This codebase marks all artifacts with a version of the format `MAJOR.MINOR.PATCH` or
+`MAJOR.MINOR.PATCH-next.ID`. For any artifact with a stable release, that is its version does not
+end in `-next.ID`, no backwards-incompatible changes will be made unless incrementing the `MINOR`
+version number. Patch releases with urgent cherry-picked bugfixes will be made by incrementing the
+`PATCH` version.
+
+A backwards-incompatible change affects the public API. The public API is any public reachable endpoint
```
Comment (christian-kreuzberger-dtx) [inline VERSIONING.MD:11]: In my opinion, public API could be a lot more then what is inferred by this document.

E.g.,

* all (http) endpoints of `keptn/distributor` (it's not just a proxy, it also used to have port 10999 in 0.10.0, which got removed in 0.11.2)
* go-utils, kubernetes-utils (this is source code! but it has functions that act as an API that is used in many integrations)
* end-user facing changes in the CLI (e.g., adding a new mandatory field to `keptn trigger delivery`)
* CloudEvents
* Internal changes that still affect the user (e.g, mongodb upgrade from 0.10.0 to 0.11.2)

My statement in this document would be:

> A breaking change could be but is not limited to
> * public REST API
> * Go Code (e.g., go-utils, kubernetes-utils, go-sdk)
> * CLI Commands
> * CloudEvents

Comment (thisthat) [reply VERSIONING.MD:11]: Thank you for your suggestions. I will adapt the PR!

**Review by odubajDT — APPROVED** [review]
LGTM

**Pull request merged**